bin/worker

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env node
+
+const mq = require("../dist/queue.js");
+const program = require("commander");
+
+program
+  .version("1.2.0")
+  .arguments("<routingKey>")
+  .description("auto reply-worker.")
+  .option("-d, --data <json>", "payload data (json string)")
+  .option("-t, --ttl <milliseconds>", "timeout in milliseconds")
+  .option("-s, --silent", "silent mode (don't output anything)")
+  .action(function(routingKey) {
+    const payload = program.data ? JSON.parse(program.data) : {};
+    const ttl = program.ttl ? parseInt(program.ttl) : undefined;
+    if (routingKey) {
+      mq.connect();
+      const interval = setInterval(() => {
+        if (mq.isConnected()) {
+          clearInterval(interval);
+          mq.worker(routingKey, () => {
```

Comment:
`mq.worker` needs to be called before `mq.isConnected()` returns true or there will be a race condition resulting in the worker never being subscribed to its queue.

Reply:
I think it is a rather surprising requirement that a worker can only be registered *before* mq connects. IMO the real fix is to remove that requirement.